cli/types/index.d.ts

```diff
@@ -2934,6 +2942,130 @@ declare namespace Cypress {
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'scrolled', fn: ($el: JQuery) => void): void
+    /**
+     * Fires when a cy command is first invoked and enqueued to be run later. Useful for debugging purposes if you're confused about the order in which commands will execute.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'command:enqueued', fn: (command: EnqueuedCommand) => void): void
+    /**
+     * Fires when cy begins actually running and executing your command. Useful for debugging and understanding how the command queue is async.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'command:start', fn: (command: CommandQueue) => void): void
+    /**
+     * Fires when cy finishes running and executing your command. Useful for debugging and understanding how commands are handled.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'command:end', fn: (command: CommandQueue) => void): void
+    /**
+     * Fires whenever a command begins its retrying routines. This is called on the trailing edge after Cypress has internally waited for the retry interval. Useful to understand **why** a command is retrying, and generally includes the actual error causing the retry to happen. When commands fail the final error is the one that actually bubbles up to fail the test. This event is essentially to debug why Cypress is failing.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'command:retry', fn: (command: CommandQueue) => void): void
+    /**
+     * Fires whenever a command emits this event so it can be displayed in the Command Log. Useful to see how internal cypress commands utilize the {% url 'Cypress.log()' cypress-log %} API.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'log:added', fn: (log: any, interactive: boolean) => void): void
+    /**
+     * Fires whenever a command's attributes changes. This event is debounced to prevent it from firing too quickly and too often. Useful to see how internal cypress commands utilize the {% url 'Cypress.log()' cypress-log %} API.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'log:changed', fn: (log: any, interactive: boolean) => void): void
+    /**
+     * Fires before the test and all **before** and **beforeEach** hooks run.
+     * @see https://docs.cypress.io/api/events/catalog-of-events.html#App-Events
+     */
+    (action: 'test:before:run', fn: (attributes: ObjectLike, test: Mocha.ITest) => void): void

```

Comment:
It was hard to tell what this object might look like. I couldn't find a clear emitter of this event. I found emitting of the `:async` version referenced in `socket.coffee`. Feel free to reference a file for more clear typing. Right now `ObjectLike` says there can by any key in the object that has a value of `any`. `Object` or `object` won't work because then it is an object with no properties.